dfastmi/io/map_file.py

```diff
@@ -0,0 +1,347 @@
+
+        Arguments
+        ---------
+        target_file : Path
+            Path to the target file.
+        """
+        target_file.unlink(missing_ok=True)
+        
+        with nc.Dataset(self._map_file) as source_dataset:
+            with nc.Dataset(target_file, "w", format="NETCDF4") as target_dataset:
+
+                mesh_variable = source_dataset.variables[self.mesh2d_name]
+
+                MapFile._copy_var(source_dataset, self.mesh2d_name, target_dataset)
+                
+                mesh_attrs = [
+                    "face_node_connectivity",
+                    "edge_node_connectivity",
+                    "edge_face_connectivity",
+                    "face_coordinates",
+                    "edge_coordinates",
+                    "node_coordinates",
+                ]
+                for mesh_attr in mesh_attrs:
+                    var_names = MapFile._get_var_names_from_var_attribute(mesh_variable, mesh_attr)
+                    for var_name in var_names:
+                        MapFile._copy_var(source_dataset, var_name, target_dataset)
+
+                        # check if variable has bounds attribute, if so copy those as well
+                        variable = source_dataset.variables[var_name]
+
+                        bounds_attr = "bounds"
+                        bounds_var_names = MapFile._get_var_names_from_var_attribute(variable, bounds_attr)
+                        for bounds_var_name in bounds_var_names:
+                            MapFile._copy_var(source_dataset, bounds_var_name, target_dataset)
+
+    @staticmethod
+    def _get_var_names_from_var_attribute(variable: nc.Variable, attr_name: str) -> List[str]:
+        try:
+            var_names = variable.getncattr(attr_name).split()
+        except AttributeError:
+            var_names = []
+            
+        return var_names
+        
+    def add_variable(
+        self,
+        variable_name: str,
+        data: np.array,
+        mesh_name: str,
+        face_dimension_name: str,
+        long_name: str,
+        unit: str,
+    ) -> None:
+        """
+        Add a new variable defined at faces to an existing UGRID netCDF file
+
+        Arguments
+        ---------
+        variable_name : str
+            Name of netCDF variable to be written.
+        data : numpy.array
+            Linear array containing the data to be written.
+        mesh_name : str
+            Name of mesh variable in the netCDF file.
+        face_dimension_name : str
+            Name of the face dimension of the selected mesh.
+        long_name : str
+            Long descriptive name for the variable ("None" if no long name attribute
+            should be written).
+        unit : str
+            String indicating the unit ("None" if no unit attribute should be written).
+        """
+        with nc.Dataset(self._map_file, "a") as dst:
+
+            var = dst.createVariable(variable_name, "f8", (face_dimension_name,))
```

Comment:
should the"f8" maybe be something that we define in one location?
To make sure it is the same encoding all over the software?

Reply:
@priscavdsluis, what are your thoughts on this?

Reply:
I checked and we only use it here. So I think we can leave it here for now .